integrations/openai/README.md

```diff
@@ -0,0 +1,152 @@
+
+#### Create an OpenAIToolkit
+[OpenAI function calling](https://platform.openai.com/docs/guides/function-calling) allows you to connect models like `gpt-4o-mini` to external tools and systems, and OpenAIToolkit provides the ability to use UC functions as tools.
+
+```python
+from unitycatalog_ai_openai.toolkit import OpenAIToolkit
+
+# create an OpenAIToolkit that include the above UC function
+toolkit = OpenAIToolkit(function_names=[f"{CATALOG}.{SCHEMA}.python_exec"])
+
+# fetch the tools stored in the toolkit
+tools = toolkit.tools
+
+# this is the function definition of the tool accepted by OpenAI
+python_exec_tool = tools[0]
+```
+
+#### Use the tools in OpenAI models
+Now we use the tools when calling OpenAI Chat Completion API.
+
+```python
+import openai
+
+messages = [
+            {
+                "role": "system",
+                "content": "You are a helpful customer support assistant. Use the supplied tools to assist the user.",
+            },
+            {"role": "user", "content": "What is the result of 2**10?"},
+        ]
+response = openai.chat.completions.create(
+                model="gpt-4o-mini",
+                messages=messages,
+                tools=tools,
+            )
+# check the model response
+print(response)
+```
+
+Handle the response and execute the function based on response result
+
+```python
+import json
+
+# there should only be one tool call
+tool_call = response.choices[0].message.tool_calls[0]
+# extract arguments
+arguments = json.loads(tool_call.function.arguments)
+
+# execute the function based on the arguments
+result = client.execute_function(func_name, arguments)
+print(result.value)
+```
+
+Construct the OpenAI response from the tool calling result
+
+```python
+# Create a message containing the result of the function call
+function_call_result_message = {
+    "role": "tool",
+    "content": json.dumps({"content": result.value}),
+    "tool_call_id": tool_call.id,
+}
+assistant_message = response.choices[0].message.to_dict()
+completion_payload = {
+    "model": "gpt-4o-mini",
+    "messages": [*messages, assistant_message, function_call_result_message],
+}
+
+# Generate final response
+openai.chat.completions.create(
+    model=completion_payload["model"], messages=completion_payload["messages"]
+)
+```
+
+#### FAQ
+
+#### What if I want to use different client for different toolkits?
+
+To use different clients during toolkit creation stage, you could pass the client directly to OpenAIToolkit:
+
+```python
+from unitycatalog_ai_openai.toolkit import OpenAIToolkit
+
+toolkit = OpenAIToolkit(function_names=[...], client=your_own_client)
+```
+
+Please note that this client is only used for retrieving UC functions so we can generate OpenAI accepted function definitions, which you could pass to the OpenAI API call. After getting a response from the OpenAI API, you should be responsible for executing them using the corresponding client with `client.execute_function(...)` as above example.
```

Comment:
can we replace this with code?

Reply:
The example above explains how to handle the response, this is just a reminder I think